### ISSUE TITLE
Invalid SemVer Range errors for git tags since semver 5.4.0 was published

### ISSUE THREAD

**Issue by zbjornson** [yarnpkg/yarn#4009]
**Do you want to request a *feature* or report a *bug*?**
potential *bug*

**What is the current behavior?**

Our `package.json` contains a few commit-ish tags, like `"uglify-js": "github:mishoo/UglifyJS2#harmony-v2.8.22"` or `"package-name#master"`. `yarn install` and `yarn install --loose-semver` both throw `An unexpected error occurred: "Invalid SemVer Range: master"` (or `harmony-v2.8.22`).

Trying `--loose-semver` came from #2070.

**If the current behavior is a bug, please provide the steps to reproduce.**

```
# cat package.json 
{
  "dependencies": {
    "uglify-js": "github:mishoo/UglifyJS2#harmony-v2.8.22"
  }
}
# yarn
yarn install v0.27.5
warning package.json: No license field
info No lockfile found.
warning No license field
[1/4] Resolving packages...
error An unexpected error occurred: "Invalid SemVer Range: harmony-v2.8.22".
```

<details>
  <summary>yarn-error.log</summary>
# cat /root/temp/yarn-error.log 
Arguments: 
  /usr/bin/nodejs /usr/bin/yarn -v

PATH: 
  /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

Yarn version: 
  0.27.5

Node version: 
  6.11.1

Platform: 
  linux x64

npm manifest: 
  {
    "dependencies": {
      "uglify-js": "github:mishoo/UglifyJS2#harmony-v2.8.22"
    }
  }

yarn manifest: 
  No manifest

Lockfile: 
  No lockfile

Trace: 
  TypeError: Invalid SemVer Range: harmony-v2.8.22
      at new Range (/usr/lib/node_modules/yarn/node_modules/semver/semver.js:776:11)
      at Function.maxSatisfying (/usr/lib/node_modules/yarn/node_modules/semver/semver.js:1156:18)
      at PackageConstraintResolver.reduce (/usr/lib/node_modules/yarn/lib/package-constraint-resolver.js:23:37)
      at Config.resolveConstraints (/usr/lib/node_modules/yarn/lib/config.js:197:36)
      at /usr/lib/node_modules/yarn/lib/util/git.js:380:35
      at next (native)
      at step (/usr/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at /usr/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14
      at Promise.F (/usr/lib/node_modules/yarn/node_modules/core-js/library/modules/_export.js:35:28)
      at /usr/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:14:12
</details>

**What is the expected behavior?**

Should work. This seems to have broken suddenly for us; 0.27.5 worked earlier this morning, so I'm not sure if one of yarn's dependencies had a breaking change introduced within its allowed semver range. In fact ... a new version of semver was published 42 minutes ago. **Edit: semver 5.3.0 works, semver 5.4.0 does not.**

**Please mention your node.js, yarn and operating system version.**

node: v6.11.1
yarn: 0.27.5
OS: Ubuntu 16.06

**Comment by zbjornson** [comment]
Corresponding node-semver issue: https://github.com/npm/node-semver/issues/201. The behavior of `maxSatisfying` changed.

**Comment by zbjornson** [comment]
Fixed in node-semver 5.4.1.
